From 06f9c409e7489a3a6a772950da453b8a008ecd3b Mon Sep 17 00:00:00 2001
From: Fabian Blatz <fabianblatz@gmail.com>
Date: Sun, 30 Nov 2025 19:19:01 +0100
Subject: [PATCH] modules: lvgl: Prevent false pointer input events

Fixes an issue where input events which have the sync flag set but are
neither X/Y coordiante updates nor press/release updates triggers a false
reporting of input to LVGL.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>
---
 modules/lvgl/input/lvgl_pointer_input.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/lvgl/input/lvgl_pointer_input.c b/modules/lvgl/input/lvgl_pointer_input.c
index 927c994499b..ea687faa0c8 100644
--- a/modules/lvgl/input/lvgl_pointer_input.c
+++ b/modules/lvgl/input/lvgl_pointer_input.c
@@ -57,6 +57,8 @@ static void lvgl_pointer_process_event(struct input_event *evt, void *user_data)
 		data->common_data.pending_event.state =
 			evt->value ? LV_INDEV_STATE_PRESSED : LV_INDEV_STATE_RELEASED;
 		break;
+	default:
+		return;
 	}
 
 	if (!evt->sync) {
-- 
2.34.1

